File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const len = computers.length;
    const visited = new Array(len).fill(false)
    
    const dfs = (i) => {
        visited[i] = true;
        
        for (let j = 0; j < len; j++) {
            if (!visited[j] && computers[i][j] == 1) {
                dfs(j);
            }
        }        
    }
    
    for (let i = 0; i < len; i++) {
        if (!visited[i]) {
            dfs(i);
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    private int[][] computers;
    private int n;
    private boolean[] v;
    private int answer = 0;
    
    public int solution(int n, int[][] computers) {
        this.n = n;
        this.computers = computers;        
        this.v = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (v[i]) continue;
            dfs(i);
            answer++;
        }
        
        for (boolean vv : v) {
            if (!vv) answer++;
        }
        
        return answer;
    }
    
    private void dfs(int cur) {    
        v[cur] = true;
        
        for (int i = 0; i < n; i++) {
            if (cur == i) continue;
            
            if (computers[cur][i] == 1 && !v[i]) {
                dfs(i);
            }
        }
    }
}